sql/table_cache.cc

```diff
@@ -649,7 +650,9 @@ void tdc_deinit(void)
     tdc_inited= false;
     lf_hash_destroy(&tdc_hash);
     mysql_mutex_destroy(&LOCK_unused_shares);
-    delete [] tc;
+    for (uint32 i= 0; i < tc_instances; i++)
+      tc[i].~Table_cache_instance();
```

Comment:
instead of using placement new and invoking destructor directly, I'd suggest to implement Table_cache_instance::new and Table_cache_instance::delete that will use aligned alloc/free

Reply:
ok. Noted